### PR TITLE
chore: template upload process revamping

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
Move from all the manual process of exporting app and filling so many details, make template upload process a 1 click action. 
Current, the step is manual. One has to export the app manually, and then upload to s3, get the APP URL and page name correct and there is a lot of room for error. All of this can be automated. 

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/cloud-services/issues/1454

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [ ] Manual

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced a new feature for publishing templates under a "use-case" category.
  - Implemented a simplified endpoint for publishing community templates.

- **Bug Fixes**
  - Streamlined the template publishing process to enhance user experience.

- **Refactor**
  - Consolidated template-related data transfer objects to a more generic naming convention.
  - Refactored the publishing methods to align with the new template categorization.

- **Documentation**
  - Updated API documentation to reflect new endpoints and usage patterns.

- **Tests**
  - Modified existing tests to accommodate changes in template DTO naming and publishing logic.

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/constants/AnalyticsEvents.java [hunk 1]
```diff
@@ -90,7 +90,9 @@ public enum AnalyticsEvents {
 
     PARTIAL_EXPORT,
 
-    COMMUNITY_TEMPLATE_PUBLISHED;
+    COMMUNITY_TEMPLATE_PUBLISHED,
+
+    APPSMITH_TEMPLATE_PUBLISHED;
```
Comment (rahulbarwal) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/constants/AnalyticsEvents.java:95]: Do we need to measure this, mostly it is going to be internal task.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCE.java [hunk 2]
```diff
@@ -26,4 +26,6 @@ Mono<ApplicationImportDTO> mergeTemplateWithApplication(
     Mono<ApplicationTemplate> getFilters();
 
     Mono<Application> publishAsCommunityTemplate(CommunityTemplateDTO resource);
+
+    Mono<Application> publishAppsmithTemplate(CommunityTemplateDTO resource);
```
Comment (rahulbarwal) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCE.java:30]: We should change the `CommunityTemplateDTO` to something generic e.g. `BaseTemplateDTO` or `TemplateDTO`

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java [hunk 3]
```diff
@@ -398,4 +403,38 @@ public Mono<Application> publishAsCommunityTemplate(CommunityTemplateDTO resourc
                             .thenReturn(application);
                 });
     }
+
+    private Mono<ApplicationTemplate> uploadAppsmithTemplateToCS(CommunityTemplateUploadDTO communityTemplate) {
+        String url = cloudServicesConfig.getBaseUrl() + "/api/v1/app-templates/upload";
```
Comment (rahulbarwal) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:408]: I remember deciding to use a new API for this.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java [hunk 4]
```diff
@@ -398,4 +403,38 @@
...
+                            FieldName.APPLICATION_ID, application.getId(),
+                            FieldName.WORKSPACE_ID, application.getWorkspaceId());
+                    return analyticsService
+                            .sendEvent(
+                                    AnalyticsEvents.APPSMITH_TEMPLATE_PUBLISHED.getEventName(),
+                                    user.getUsername(),
+                                    data)
+                            .thenReturn(application);
+                });
+    }
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:439]: Agree with this.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java (hunk 3 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:408]: This is till pending as the changes for CS is WIP

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:439]: Although this makes sense now, its not true. The reason is i have not updated the params required for Appsmith template from the TemplateUpload app. Its still WIP, once i do that this does not make sense.

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**